app/models/question.rb

```diff
@@ -0,0 +1,57 @@
+class Question < ActiveRecord::Base
+  belongs_to :creator, class_name: "User"
+  has_many :responses, as: :respondable, dependent: :destroy
+  has_many :answers, dependent: :destroy
+  has_many :votes, as: :votable, dependent: :destroy
+  has_many :question_tags, dependent: :destroy
+  has_many :tags, through: :question_tags
+
+  def self.by_vote_sum
+    all.to_a.sort_by(&:vote_sum).reverse
+  end
+
+  def num_answers
+    self.answers.count
+  end
+
+  def self.by_recency
+    order(created_at: :desc)
+  end
+
+  def editable_by? user
+    creator == user
+  end
+
+  def vote_sum
+    Vote.where("votable_type = ? AND votable_id =?", "Question", self.id).pluck("value").inject(0){|sum,x| sum + x }
+  end
+
+  def already_voted?(user_id)
+    !Vote.where("votable_type = ? AND user_id = ? AND votable_id = ?", "Question", user_id, self.id).empty?
```

Comment:
Once you refactor these to use AR directly, they'll be identical across Q and A models - so you can pull them out into a module called Voting and mix it into those models.